root/utility/generateRelationshipAppearancesList.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import linkedEntities from '../static/scripts/common/linkedEntities';
+
+import filterRelationshipsByType from './filterRelationshipsByType';
+
+const pickAppearancesTypes = (entityType) => {
+  switch (entityType) {
+    case 'artist':
+    case 'label':
+    case 'place': {
+      return ['recording', 'release', 'release_group', 'work'];
+    }
+    case 'work': {
+      return ['recording'];
```

Comment:
Minor nitpick: can we make these two arrays constants outside of `pickAppearancesTypes`, and return the constants here? That'll avoid at least one unnecessary allocation each time the template renders. :)